src/cert.rs

```diff
@@ -95,16 +95,18 @@ pub(crate) fn parse_cert_internal<'a>(
             subject_alt_name: None,
         };
 
-        // mozilla::pkix allows the extensions to be omitted. However, since
-        // the subjectAltName extension is mandatory, the extensions are
-        // mandatory too, and we enforce that. Also, mozilla::pkix includes
+        // mozilla::pkix allows the extensions to be omitted. It also includes
         // special logic for handling critical Netscape Cert Type extensions.
         // That has been intentionally omitted.
 
+        if tbs.at_end() {
+            return Ok(cert)
```

Comment:
Instead of adding a new return branch here, let's wrap the `der::nested(` call below in an `if !tbs.at_end() {`.